internal/worldstate/leveldb/commit_and_query.go

```diff
@@ -163,6 +165,84 @@ func (l *LevelDB) GetIndexDefinition(dbName string) ([]byte, *types.Metadata, er
 	return l.Get(worldstate.DatabasesDBName, dbName)
 }
 
+// Iterator iterates over a range of keys
+type Iterator struct {
+	iterator.Iterator
+}
+
+// GetIterator returns an iterator to fetch values associated with a range of keys
+// startKey is inclusive while the endKey is exclusive
+func (l *LevelDB) GetIterator(dbName string, startKey, endKey string) (worldstate.Iterator, error) {
+	l.dbsList.RLock()
+	db := l.dbs[dbName]
+	l.dbsList.RUnlock()
+
+	if db == nil {
+		l.logger.Errorf("database %s does not exist", dbName)
+		return nil, errors.Errorf("database %s does not exist", dbName)
+	}
+
+	r := &util.Range{}
+	if startKey == "" {
+		r.Start = nil
+	} else {
+		r.Start = []byte(startKey)
+	}
+
+	if endKey == "" {
+		r.Limit = nil
+	} else {
+		r.Limit = []byte(endKey)
+	}
+
+	return &Iterator{
+		db.file.NewIterator(r, &opt.ReadOptions{}),
+	}, nil
+}
+
+// Key returns the key of the current key/value pair, or nil if done.
+// The caller should not modify the contents of the returned slice, and
+// its contents may change on the next call to any 'seeks method'.
+func (i *Iterator) Key() string {
+	return string(i.Iterator.Key())
+}
+
+// ValueAndMetadata returns the value and metadata of the current key/value
+// pair, nil if done.
+// The caller should not modify the contents of the returned slice, and
+// its contents may change on the next call to any 'seeks method'.
+func (i *Iterator) ValueAndMetadata() ([]byte, *types.Metadata, error) {
```

Comment:
why not return `*ValueAndMetadata` and let the user decide what to use?

Reply:
`*ValueAndMetadata` is an internal structure. It is exported because of the proto message.

Reply:
Done